code/infiltrate/infiltrate.go

```diff
@@ -48,43 +44,87 @@
 //
 // About logging: Infiltrate logs everything to the syslog.  Errors encountered during startup are
 // also logged to stderr.
+//
+// Services and input formats:
+//
+// /sonar-reading
+//    Input is JSON-format Sonar monitoring data: array of go-utils/sonarlog.SonarReading.  Input
+//    fields are checked against that record type: fields that are not known to the type are
+//    dropped.
+//
+// /sonar-heartbeat
+//    Input is JSON-format Sonar heartbeat data: array of go-utils/sonarlog.SonarHeartbeat.  Input
+//    fields are checked against that record type: fields that are not known to the type are
+//    dropped.
+//
+// /sonar-freecsv?cluster=clusterName
+//    Input is "free CSV" format Sonar monitoring and heartbeat data, intermixed, one record per
+//    line.  If -match-user-and-cluster is used then the the user name provided by the HTTP
+//    connection must match the "cluster" parameter, which is always required.  Input fields are not
+//    checked - the records are always stored verbatim in the database, after checking the cluster.
+//
+// /sysinfo?cluster=clusterName
+//    Input is JSON-format system information data: a single record of go-utils/config.SystemConfig.
+//    The cluster parameter is required; cluster name checking is as for /sonar-freecsv.
```

Comment:
The record must have sensible `timestamp` and `hostname` fields, or they will be rejected.